libavcodec/decode.c

```diff
@@ -1367,6 +1367,53 @@ static void hwaccel_uninit(AVCodecContext *avctx)
     av_buffer_unref(&avctx->hw_frames_ctx);
 }
 
+static const AVCodecHWConfigInternal *get_hw_config(AVCodecContext *avctx, enum AVPixelFormat fmt)
+{
+    const AVCodecHWConfigInternal *hw_config;
+    int i;
+
+    if (avctx->codec->hw_configs) {
+        for (i = 0;; i++) {
+            hw_config = avctx->codec->hw_configs[i];
+            if (!hw_config)
+                break;
+            if (hw_config->public.pix_fmt == fmt)
+                break;
+        }
+    } else {
+        hw_config = NULL;
+    }
+
+    return hw_config;
+}
+
+static int hwaccel_realloc_surface(AVCodecContext *avctx)
+{
+    const AVCodecHWConfigInternal *hw_config;
+    int ret;
+
+    if (avctx->hw_frames_ctx)
+        av_buffer_unref(&avctx->hw_frames_ctx);
+
+    hw_config = get_hw_config(avctx, avctx->pix_fmt);
+    if (!hw_config)
+        return AV_PIX_FMT_NONE;
+
+    if (avctx->hw_device_ctx &&
```

Comment:
there are many if at https://github.com/intel-media-ci/ffmpeg/pull/153/files#diff-ea4f4194be6309231a5b01e350c272cdL1448, why we only handle this?